File: solutions/problem_0371_sum_of_two_integers/solution.cpp
/*
 * Problem 371: Sum of Two Integers
 * ===============================
 * Difficulty: Medium
 * Tags: Math, Bit Manipulation
 * Pattern: Bit Manipulation
 *
 * Time Complexity:  O(n) or O(log n)
 * Space Complexity: O(1)
 */

#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    int getSum(int a, int b) {
        // Bit manipulation - O(n) time, O(1) space
        int result = 0;
        for (int val : a) {
            result ^= val;
        }
        return result;
    }
};

File: solutions/problem_0371_sum_of_two_integers/solution.py
"""
Problem 371: Sum of Two Integers
===============================
Difficulty: Medium
Tags: Math, Bit Manipulation
Pattern: Bit Manipulation

Time Complexity:  O(n) or O(log n)
Space Complexity: O(1)
"""

class Solution:
    def getSum(self, a: int, b: int) -> int:
        # Bit manipulation - O(n) time, O(1) space
        result = 0
        for val in a:
            result ^= val
        return result
